File: sdk/src/main/java/jp/ac/nagoya_u/dsmoq/sdk/util/InputValidationException.java
package jp.ac.nagoya_u.dsmoq.sdk.util;

import java.util.ArrayList;
import java.util.List;

/**
 * 入力値が不正であることを表す例外
 */
public class InputValidationException extends RuntimeException {
    private List<InputValidationError> errors;
    public InputValidationException(List<InputValidationError> errors) {
        this.errors = new ArrayList<>(errors);
    }

    public List<InputValidationError> getErrorMessages() {
        return new ArrayList<>(this.errors);
    }
}

File: server/apiServer/src/main/scala/dsmoq/exceptions/InputValidationException.scala
package dsmoq.exceptions

/**
 * 入力チェックに違反した場合に送出する例外
 *
 * @param errors エラーメッセージのリスト
 */
class InputValidationException(errors: Iterable[(String, String)]) extends RuntimeException {
  val validationErrors = errors

  /**
   * エラーメッセージを取得する
   */
  def getErrorMessage(): Iterable[InputValidationError] = {
    validationErrors.map {
      case (name, message) => {
        InputValidationError(
          name = name,
          message = message
        )
      }
    }
  }
}

/**
 * 入力チェックエラーのケースクラス
 *
 * @param name たいしょうのなまえ
 * @param message エラーメッセージ
 */
case class InputValidationError(
  name: String,
  message: String
)
